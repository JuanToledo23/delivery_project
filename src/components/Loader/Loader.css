.loading {
  position: fixed;
  z-index: 999;
  height: 2em;
  width: 2em;
  overflow: show;
  margin: auto;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
}

.loading > img {
  position: fixed;
  left: 50%;
  top: 50%;
  transform: translate(-50%, -50%);
}

.loading:before {
  content: '';
  display: block;
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: #00000059;
}

.loading:not(:required) {
  font: 0/0 a;
  color: transparent;
  text-shadow: none;
  background-color: transparent;
  border: 0;
}
