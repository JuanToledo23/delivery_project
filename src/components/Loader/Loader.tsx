import loaderImg from 'assets/img/spinning-circles.svg';
import './Loader.css';

function Loader() {
  
  return (
    <div>
      <div className="loading">
        <img src={loaderImg} alt="loader" />
      </div>
    </div>
  );
}

export default Loader;
