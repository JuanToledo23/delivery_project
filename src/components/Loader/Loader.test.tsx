import { render } from '@testing-library/react';
import Loader from './Loader';

describe('Loader', () => {
  it('should match snapshot', () => {
    const { asFragment } = render(<Loader />);

    expect(asFragment()).toMatchSnapshot();
  });
});
