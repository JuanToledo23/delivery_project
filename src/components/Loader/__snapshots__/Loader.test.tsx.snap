// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="loading"
    >
      <img
        alt="loader"
        src="spinning-circles.svg"
      />
    </div>
  </div>
</DocumentFragment>
`;
